src/Yi/Braid.hs

```diff
@@ -0,0 +1,598 @@
+{-# language BangPatterns #-}
+{-# language DeriveDataTypeable #-}
+{-# language LambdaCase #-}
+{-# language MultiParamTypeClasses #-}
+{-# language OverloadedStrings #-}
+{-# language ScopedTypeVariables #-}
+{-# language ViewPatterns #-}
+{-# language FlexibleContexts #-}
+{-# language FlexibleInstances #-}
+{-# language TypeFamilies #-}
+{-# language UndecidableInstances #-}
```

Comment:
This is unfortunate; it's required for the Eq, Ord, etc. instances. I don't really see why they're undecidable so it's possible I'm missing something and this could be removed somehow.